flake/default.nix

```diff
@@ -1,14 +1,66 @@
+{
+  lib,
+  config,
+  inputs,
+  partitionStack,
+  ...
+}:
+let
+  # Specify which outputs are defined by which partitions
+  partitionedAttrs = builtins.mapAttrs (_: lib.toList) {
+    checks = [
+      "dev"
+      "docs"
+    ];
+    devShells = "dev";
+    formatter = "dev";
+    packages = [
+      "dev"
+
+      # FIXME: man-docs is used by the nixvimConfiguration,
+      # which probably makes the `docs` partition pointless
+      "docs"
+    ];
+  };
+in
 {
   imports = [
-    ./dev
     ./flake-modules
     ./lib.nix
     ./legacy-packages.nix
     ./nixvim-configurations.nix
     ./overlays.nix
-    ./packages.nix
     ./templates.nix
-    ./tests.nix
     ./wrappers.nix
+    inputs.flake-parts.flakeModules.partitions
   ];
+
+  # Define flake partitions
+  # Each has a `module`, assigned to the partition's submodule,
+  # and an `extraInputsFlake`, used for its inputs.
+  # See https://flake.parts/options/flake-parts-partitions.html
+  partitions = {
+    dev = {
+      module = ./dev;
+      extraInputsFlake = ./dev;
+    };
+    docs = {
+      module = ../docs/flake-module.nix;
+      extraInputsFlake = ../docs;
+    };
+  };
+
+  # `partitionedAttrs` only supports one partition per output.
+  # We want `checks` to be defined by both `dev` and `docs`.
+  # See upstream PR: https://github.com/hercules-ci/flake-parts/pull/272
+  flake = lib.optionalAttrs (partitionStack == [ ]) (
+    builtins.mapAttrs (
+      attrName: partitions:
+      lib.mkMerge (
+        builtins.map (
+          partition: lib.mkForce config.partitions.${partition}.module.flake.${attrName}
+        ) partitions
+      )
+    ) partitionedAttrs
+  );
```

Comment:
This is the implementation from https://github.com/hercules-ci/flake-parts/pull/272, if that gets merged before this we can bump our flake-parts input and drop this.